src/interfaces/traits.rs

```diff
@@ -0,0 +1,12 @@
+use crate::errors::TockloaderError;
+use enum_dispatch::enum_dispatch;
+
+#[enum_dispatch]
+pub trait BoardInterface {
```

Comment:
While the file in tockloader is called board_interface.py (and has been for a long time), in the tockloader.py code I've switched to effectively calling this a "channel". So perhaps renaming this to `Channel` or `BoardChannel` might resolve some of the headaches with overloading the word "interface".

Reply:
Yep, I 100% agree with you on this one. I like the name `Channel` way better. The name of the `Channel` enum and `BoardChannel` trait still kind of overlap each other, but that's mostly because I don't know yet how I want to structure the traits, and what behaviour is common between the channels.